src/org/labkey/test/tests/SampleTypeLinkToStudyTest.java

```diff
@@ -0,0 +1,194 @@
+        init.doSetup();
+    }
+
+    @Override
+    protected @Nullable String getProjectName()
+    {
+        return SAMPLE_TYPE_PROJECT;
+    }
+
+    @Override
+    public List<String> getAssociatedModules()
+    {
+        return null;
+    }
+
+    @Override
+    protected void doCleanup(boolean afterTest) throws TestTimeoutException
+    {
+        _containerHelper.deleteProject(SAMPLE_TYPE_PROJECT, afterTest);
+        _containerHelper.deleteProject(LINKED_STUDY_PROJECT, afterTest);
+    }
+
+    private void doSetup() throws IOException, CommandException
+    {
+        _containerHelper.createProject(getProjectName(), null);
+        _containerHelper.createProject(LINKED_STUDY_PROJECT, "Study");
+        clickButton("Create Study");
+        clickButton("Create Study");
+
+        goToProjectHome(SAMPLE_TYPE_PROJECT);
+        new PortalHelper(getDriver()).addBodyWebPart("Sample Types");
+
+        goToProjectHome(LINKED_STUDY_PROJECT);
+        new PortalHelper(getDriver()).addBodyWebPart("Datasets");
+
+        createSampleTypes();
+    }
+
+    private void createSampleTypes() throws IOException, CommandException
+    {
+        goToProjectHome(SAMPLE_TYPE_PROJECT);
+
+        log("Creating sample types");
+        FieldDefinition.LookupInfo lookupInfo = new FieldDefinition.LookupInfo(getProjectName(), "exp.materials", SAMPLE_TYPE1);
+        TestDataGenerator dgen = new TestDataGenerator(lookupInfo)
+                .withColumns(List.of(
+                        new FieldDefinition("name", FieldDefinition.ColumnType.String)));
+        dgen.createDomain(createDefaultConnection(), SAMPLE_TYPE_DOMAIN_KIND);
+        dgen.addCustomRow(Map.of("name", "blood"));
+        dgen.addCustomRow(Map.of("name", "urine"));
+        dgen.addCustomRow(Map.of("name", "stool"));
+        dgen.addCustomRow(Map.of("name", "sweat"));
+        dgen.insertRows(createDefaultConnection(), dgen.getRows());
+
+        lookupInfo = new FieldDefinition.LookupInfo(getProjectName(), "exp.materials", SAMPLE_TYPE2);
+        dgen = new TestDataGenerator(lookupInfo)
+                .withColumns(List.of(
+                        new FieldDefinition("name", FieldDefinition.ColumnType.String),
+                        new FieldDefinition("int", FieldDefinition.ColumnType.Integer)));
+        dgen.createDomain(createDefaultConnection(), SAMPLE_TYPE_DOMAIN_KIND);
+        dgen.addCustomRow(Map.of("name", "first"));
+        dgen.addCustomRow(Map.of("name", "second"));
+        dgen.addCustomRow(Map.of("name", "third"));
+        dgen.addCustomRow(Map.of("name", "fourth"));
+        dgen.insertRows(createDefaultConnection(), dgen.getRows());
+
+    }
+
+    @Test
+    public void testLinkToStudy()
+    {
+        goToProjectHome(SAMPLE_TYPE_PROJECT);
+
+        log("Linking sample types one row to study");
+        linkToStudy(SAMPLE_TYPE1, 2, Arrays.asList("P1", "P2"), Arrays.asList("1", "2"));
```

Comment:
We should also test setting up the source samples with fields of types : subject & visit (ID or date) as well. I think it's fine to manually specify the subject/timepoint info in the confirm view, but we also want to verify that we can use the new datatypes to identify this information.

Reply:
Done